environs/interface.go

```diff
@@ -136,6 +136,20 @@ type ProviderCredentials interface {
 	) (*cloud.Credential, error)
 }
 
+// ProviderCredentialsCloudSpecific is an interface that an EnvironProvider implements
+// in order to validate and automatically detect credentials for a specific cloud
+// supported by the provider.
+type ProviderCredentialsCloudSpecific interface {
+	// DetectCredentialsFor automatically detects one or more credentials
+	// from the environment. This may involve, for example, inspecting
+	// environment variables, or reading configuration files in
+	// well-defined locations.
+	//
+	// If no credentials can be detected, DetectCredentials should
+	// return an error satisfying errors.IsNotFound.
+	DetectCredentialsFor(cloudName string) (*cloud.CloudCredential, error)
```

Comment:
Rather than a new method, we could also just make the existing `DetectCredentials()` method on the `ProviderCredentials` interface take a cloud name. The cloud name would be ignored except when it's needed. This would be cleaner as it avoids an extra (unnecessary) interface and method. This also then reuses the validation logic to check for multiple credentials etc.